source/config/SettingConfig.h

```diff
@@ -235,6 +235,19 @@ namespace emp {
       return new_ptr->value;
     }
 
+    /// Add a new setting not linked to a variable
+
+    template <typename T>
+    T & AddSetting(const std::string & name,
+                   const std::string & desc="",
+                   const char option_flag='\0')
+    {
+      emp_assert(!emp::Has(setting_map, name));
```

Comment:
Could we default construct a `T t;` and then just call the fun above so we just have one impl?

Reply:
The issue with that is that then we end up with dynamically allocated memory in some cases but not all. We could take care of that with a flag, but since this file is deprecated anyway I don't know if it's worth bothering

Reply:
I'm pretty sure we would still always end up with dynamically allocated memory? The function we're delegating to always `new`s
https://github.com/devosoft/Empirical/blob/6a4b844ddf0b3e193bd25827853745203425d39a/source/config/SettingConfig.h#L233

What I'm thinking of would look like
```

    template <typename T>
    T & AddSetting(const std::string & name,
                   const std::string & desc="",
                   const char option_flag='\0')
    {
      T t;
      return AddSetting(name, desc, option_flag, t);
    }
```
    }

Reply:
Sorry, I didn't explain that clearly. We can't default construct a T because it's passed by reference and if we construct it here it will immediately go out of scope. The way to get around that is to dynamically allocate memory for the value inside the SettingInfo constructor. That's the additional dynamically allocated memory I was referring to.